plugins/inputs/t128_graphql/response_processor.go

```diff
@@ -0,0 +1,66 @@
+package t128_graphql
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/Jeffail/gabs"
+)
+
+//ProcessedResponse stores the processed fields and tags for injection into telegraf accumulator
+type ProcessedResponse struct {
+	Fields map[string]interface{}
+	Tags   map[string]string
+}
+
+// ProcessResponse takes in a query response, pulls out the desired data and stores it in a struct
+func ProcessResponse(jsonChildren []*gabs.Container, collectorName string, fields map[string]string, tags map[string]string) ([]*ProcessedResponse, error) {
+	var processedResponses []*ProcessedResponse
+
+	//TODO: support queries needed by mist agent - MON-315
+	for _, child := range jsonChildren {
+		node := child.Data().(map[string]interface{})
+		processedResponse := &ProcessedResponse{Fields: map[string]interface{}{}, Tags: map[string]string{}}
```

Comment:
NBD, but one pattern that I see a lot is to put this initialization in a `New` method so that this is a little cleaner. It is usually also helpful if there is other initialization needed.
```
func NewProcessedResponse() ProcessedResponse {
    return ProcessedResponse{Fields: map[string]interface{}{}, Tags: map[string]string{}}
}
```